src/BrowserFetcher.ts

```diff
@@ -220,14 +220,29 @@ export class BrowserFetcher {
     if (!(await existsAsync(this._downloadsFolder)))
       await mkdirAsync(this._downloadsFolder);
     try {
-      await downloadFile(url, archivePath, progressCallback);
-      await install(archivePath, outputPath);
+      if (os.arch() === 'arm64') {
+        fs.stat('/usr/bin/chromium-browser', function (err, stats) {
+          if (stats === undefined) {
+            console.error(` The chromium binary is not available for arm64: `);
+            console.error(` If you are on Ubuntu,You can install with: `);
+            console.error(` \n apt-get install chromium-browser\n`);
+            throw new Error();
+          }
+        });
+      } else {
+        await downloadFile(url, archivePath, progressCallback);
+        await install(archivePath, outputPath);
+      }
     } finally {
-      if (await existsAsync(archivePath)) await unlinkAsync(archivePath);
+      if (os.arch() !== 'arm64')
+        if (await existsAsync(archivePath)) await unlinkAsync(archivePath);
     }
     const revisionInfo = this.revisionInfo(revision);
-    if (revisionInfo) await chmodAsync(revisionInfo.executablePath, 0o755);
-    return revisionInfo;
+    if (revisionInfo) {
+      if (os.arch() !== 'arm64')
+        await chmodAsync(revisionInfo.executablePath, 0o755);
+      return revisionInfo;
+    }
```

Comment:
I'm a bit concerned about all the different conditionals that now exist around the `os.arch()` check. I wonder if there's a nicer way to just completely bypass any of this if we're on `arm64` ?

Reply:
@jackfranklin I will be happy to modify the code more please do revert and provide your suggestions on to what do you exactly want to replace ```os.arch()``` with.

Reply:
Let's indeed keep the arm64-specific logic in a single place. Something like this:

```js
if (os.arch() === 'arm64') {
  await handleArm64();
  return;
}
// existing code for non-arm64 goes here...
```

Reply:
@mathiasbynens incorporated the changes you suggested, please do revert if something else is required to be done.